src/ImageSharp/Formats/Gif/GifDecoderCore.cs

```diff
@@ -250,33 +251,50 @@ private void ReadLogicalScreenDescriptor()
         }
 
         /// <summary>
-        /// Reads the application extension block parsing any animation information
+        /// Reads the application extension block parsing any animation or XMP information
         /// if present.
         /// </summary>
         private void ReadApplicationExtension()
         {
             int appLength = this.stream.ReadByte();
 
             // If the length is 11 then it's a valid extension and most likely
-            // a NETSCAPE or ANIMEXTS extension. We want the loop count from this.
+            // a NETSCAPE, XMP or ANIMEXTS extension. We want the loop count from this.
             if (appLength == GifConstants.ApplicationBlockSize)
             {
-                this.stream.Skip(appLength);
-                int subBlockSize = this.stream.ReadByte();
+                this.stream.Read(this.buffer, 0, GifConstants.ApplicationBlockSize);
+                bool isXmp = true;
+                ReadOnlySpan<byte> idBytes = GifConstants.XmpApplicationIdentificationBytes;
+                for (int i = 0; i < idBytes.Length; i++)
+                {
+                    isXmp &= this.buffer[i] == idBytes[i];
+                }
```

Comment:
```suggestion
                bool isXmp = buffer.AsSpan().StartsWith(idBytes);
```
?

`StartsWith` is optimized (and vectorized).

Reply:
Good suggestion